libraries/src/Application/MultiFactorAuthenticationHandler.php

```diff
@@ -311,6 +311,12 @@ private function needsMultiFactorAuthenticationRedirection(): bool
 			return false;
 		}
 
+		// Allow the Joomla update finalisation to run
+		if ($isAdmin && $option === 'com_joomlaupdate' && in_array($task, ['update.finalise', 'update.cleanup']))
```

Comment:
```suggestion
		if ($isAdmin && $option === 'com_joomlaupdate' && in_array($task, ['update.finalise', 'update.cleanup', 'update.finaliseconfirm']))
```

Isnt [finalisecofirm ](https://github.com/joomla/joomla-cms/blob/4.2-dev/administrator/components/com_joomlaupdate/src/Controller/UpdateController.php#L433) also to be allowed in some cases? Or I'm wrong here?

Reply:
Good catch